homeassistant/components/airvisual_pro/config_flow.py

```diff
@@ -19,48 +33,86 @@
 )
 
 
+async def async_validate_credentials(ip_address: str, password: str) -> dict[str, Any]:
+    """Validate an IP address/password combo (and return any errors as appropriate)."""
+    node = NodeSamba(ip_address, password)
+    errors = {}
+
+    try:
+        await node.async_connect()
+    except InvalidAuthenticationError as err:
+        LOGGER.error("Invalid password for Pro at IP address %s: %s", ip_address, err)
+        errors["base"] = "invalid_auth"
+    except NodeConnectionError as err:
+        LOGGER.error("Cannot connect to Pro at IP address %s: %s", ip_address, err)
+        errors["base"] = "cannot_connect"
+    except NodeProError as err:
+        LOGGER.error("Unknown Pro error while connecting to %s: %s", ip_address, err)
+        errors["base"] = "unknown"
+    except Exception as err:  # pylint: disable=broad-except
+        LOGGER.exception("Unknown error while connecting to %s: %s", ip_address, err)
+        errors["base"] = "unknown"
+    finally:
+        await node.async_disconnect()
+
+    return errors
+
+
 class AirVisualProFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle an AirVisual Pro config flow."""
 
     VERSION = 1
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        return await self.async_step_reauth_confirm()
```

Comment:
It would be a bit safer design if you saved the config entry here in case `self.context['entry_id']` ever gets overwritten.  I'm seen that happen in other cases but sadly I can't remember the exact circumstances